libraries/bot-dialogs/src/main/java/com/microsoft/bot/dialogs/WaterfallDialog.java

```diff
@@ -267,8 +267,8 @@ private String waterfallStepName(int index) {
         String stepName = steps.get(index).getClass().getSimpleName();
 
         // Default stepname for lambdas
-        if (StringUtils.isAllBlank(stepName) || stepName.contains("<")) {
-            stepName = String.format("Step%0of%1", index + 1, steps.size());
+        if (StringUtils.isAllBlank(stepName) || stepName.contains("$Lambda$")) {
```

Comment:
I don't understand this change, C# [checks](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs/WaterfallDialog.cs#L263) for `<`

Reply:
What `steps.get(index).getClass().getSimpleName();` returns in C# and Java is completely different.

Reply:
This probably was never tested or used, and the modification in line 271 support this theory. That line failed because the format was wrong, and if this issue wasn't found earlier, its because either the method or the `if` was never called or true respectively.

Reply:
JS is also [different](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder-dialogs/src/waterfallDialog.ts#L334). This is weird, it will affect to several tests

Reply:
I think that functionality is very specific for C#, at least the `stepName.contains("<")`. The `stepName.contains("$Lambda$")` might be removed and just validate that the steps contains something (just leaving the `isAllBlank`)